jOOQ/src/main/java/org/jooq/tools/LoggerListener.java

```diff
@@ -82,91 +84,121 @@
 
     private static final JooqLogger log   = JooqLogger.getLogger(LoggerListener.class);
 
+    private final Log.Level executeLoggingBindVariables;
+
+    private final boolean   executeLoggingAbbreviatedBindVariables;
+
+    private final Log.Level executeLoggingSqlString;
+
+    private final Log.Level executeLoggingResult;
+
+    private final int       executeLoggingResultNumberOfRows;
+
+    private final int       executeLoggingResultNumberOfColumns;
+
+    private final Log.Level executeLoggingRoutine;
+
+    private final Log.Level executeLoggingException;
+
+    public LoggerListener(final Log.Level executeLoggingBindVariables,
+                          final Boolean executeLoggingAbbreviatedBindVariables,
+                          final Log.Level executeLoggingSqlString,
+                          final Log.Level executeLoggingResult,
+                          final Integer executeLoggingResultNumberOfRows,
+                          final Integer executeLoggingResultNumberOfColumns,
+                          final Log.Level executeLoggingRoutine,
+                          final Log.Level executeLoggingException) {
+        this.executeLoggingBindVariables = requireNonNull(executeLoggingBindVariables);
+        this.executeLoggingAbbreviatedBindVariables = requireNonNull(executeLoggingAbbreviatedBindVariables);
+        this.executeLoggingSqlString = requireNonNull(executeLoggingSqlString);
+        this.executeLoggingResult = requireNonNull(executeLoggingResult);
+        this.executeLoggingResultNumberOfRows = requireNonNull(executeLoggingResultNumberOfRows);
+        this.executeLoggingResultNumberOfColumns = requireNonNull(executeLoggingResultNumberOfColumns);
+        this.executeLoggingRoutine = requireNonNull(executeLoggingRoutine);
+        this.executeLoggingException = requireNonNull(executeLoggingException);
+    }
+
     @Override
     public void renderEnd(ExecuteContext ctx) {
-        if (log.isDebugEnabled()) {
-            Configuration configuration = ctx.configuration();
-            String newline = TRUE.equals(configuration.settings().isRenderFormatted()) ? "\n" : "";
+        Configuration configuration = ctx.configuration();
+        String newline = TRUE.equals(configuration.settings().isRenderFormatted()) ? "\n" : "";
 
-            // [#2939] Prevent excessive logging of bind variables only in DEBUG mode, not in TRACE mode.
-            if (!log.isTraceEnabled())
-                configuration = abbreviateBindVariables(configuration);
+        // [#2939] Prevent excessive logging of bind variables only in DEBUG mode, not in TRACE mode.
+        if (executeLoggingAbbreviatedBindVariables)
+            configuration = abbreviateBindVariables(configuration);
 
-            String[] batchSQL = ctx.batchSQL();
-            if (ctx.query() != null) {
+        String[] batchSQL = ctx.batchSQL();
+        if (log.isEnabled(executeLoggingSqlString) && ctx.query() != null) {
```

Comment:
I suggest you inverse these checks. The `null` check is much cheaper than the enabled check, which can get quite hairy, checking several `volatile` fields, depending on the logging provider.